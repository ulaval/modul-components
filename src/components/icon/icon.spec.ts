import { createLocalVue, mount } from '@vue/test-utils';
import Vue, { VueConstructor } from 'vue';

import { renderComponent } from '../../../tests/helpers/render';
import IconPlugin, { MIcon } from './icon';

describe('MIcon', () => {
    let localVue: VueConstructor<Vue>;

    beforeEach(() => {
        localVue = createLocalVue();
        localVue.use(IconPlugin);
    });

    it('should render correctly', () => {
        const icon = mount(MIcon, {
            localVue: localVue,
            propsData: {
                name: 'options',
                size: '20px'
            }
        });

        return expect(renderComponent(icon.vm)).resolves.toMatchSnapshot();
    });

    it('should render title when svgTitle prop is set', () => {
        const icon = mount(MIcon, {
            localVue: localVue,
            propsData: {
                svgTitle: 'options'
            }
        });

        return expect(renderComponent(icon.vm)).resolves.toMatchSnapshot();
    });

    it('should emit click event when icon is clicked', () => {
        const icon = mount(MIcon, {
            localVue: localVue
        });

        icon.trigger('click');

        expect(icon.emitted('click')).toBeTruthy();
    });
});
