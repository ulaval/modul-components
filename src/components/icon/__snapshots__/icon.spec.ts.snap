// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MIcon should render correctly 1`] = `
<svg aria-hidden="true" width="20px" height="20px" class="m-icon">
    <use xlink:href="#options" aria-hidden="true"></use>
</svg>
`;

exports[`MIcon should render title when svgTitle prop is set 1`] = `
<svg width="1em" height="1em" class="m-icon">
    <title>options</title>
    <use xlink:href="#default" aria-hidden="true"></use>
</svg>
`;
